File: stress/t/arr_matvec.f
      PROGRAM ARRMATVEC
      INTEGER*8 N,R,C,ACC,A,X,T,S
      INTEGER*8 M(0:35),V(0:5),Y(0:5)
      N=6
      R=0
   10 IF (R.GE.N) GOTO 30
      V(R)=R+1
      C=0
   20 IF (C.GE.N) GOTO 25
      M(R*N+C)=(R+1)*(C+1)-R
      C=C+1
      GOTO 20
   25 CONTINUE
      R=R+1
      GOTO 10
   30 CONTINUE
      R=0
   40 IF (R.GE.N) GOTO 60
      ACC=0
      C=0
   50 IF (C.GE.N) GOTO 55
      A=M(R*N+C)
      X=V(C)
      ACC=ACC+A*X
      C=C+1
      GOTO 50
   55 CONTINUE
      Y(R)=ACC
      R=R+1
      GOTO 40
   60 CONTINUE
      S=0
      R=0
   70 IF (R.GE.N) GOTO 80
      T=Y(R)
      S=S+T
      R=R+1
      GOTO 70
   80 CONTINUE
      WRITE(*,'(A,I0)') 'ysum=',S
      T=Y(0)
      WRITE(*,'(A,I0)') 'y0=',T
      T=Y(5)
      WRITE(*,'(A,I0)') 'y5=',T
      END

File: stress/t/arr_matvec.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t n,r,c,acc,a,x,t,s;
  int64_t M[36],V[6],Y[6];
  n=6;
  for(r=0;r<n;r++){
    V[r]=r+1;
    for(c=0;c<n;c++){ M[r*n+c]=(r+1)*(c+1)-r; }
  }
  for(r=0;r<n;r++){
    acc=0;
    for(c=0;c<n;c++){ a=M[r*n+c]; x=V[c]; acc=acc+a*x; }
    Y[r]=acc;
  }
  s=0;
  for(r=0;r<n;r++){ t=Y[r]; s=s+t; }
  printf("ysum="); printf("%ld",(long)s); printf("\n");
  t=Y[0]; printf("y0="); printf("%ld",(long)t); printf("\n");
  t=Y[5]; printf("y5="); printf("%ld",(long)t); printf("\n");
  return 0;
}
